### PR TITLE
Use enum for pixel shapes

### PR DESCRIPTION
**Pull request by maxnoe**
When preparing to update the bokeh event viewer according to #1330 and #1247, 
I found working with these strings very clumsy.

Replaced the string with an enum, which should make many things much more straight forward.

### CODE REVIEW

**Comment by kosack** [comment]
Is this really more straightforward? It is right now text metadata, so very human readable, and this changes the data model. I don't see a big advantage, other than avoiding a single string compare (we don't support different pixel types being mixed in a camera)

**Comment by kosack** [comment]
I also see no reason to support "circle" shapes - we always have Winston cones, so the actual area is a hexagon.

File: ctapipe/instrument/camera/geometry.py [hunk 1]
```diff
@@ -371,7 +419,7 @@ def to_table(self):
             [self.pix_id, self.pix_x, self.pix_y, self.pix_area],
             names=["pix_id", "pix_x", "pix_y", "pix_area"],
             meta=dict(
-                PIX_TYPE=self.pix_type,
+                PIX_TYPE=self.pix_type.value,
```
Comment (kosack) [inline ctapipe/instrument/camera/geometry.py:422]: again, this is not very readable the user has to know what pixel type this integer corresponds to - why not just use the string representation and then it doesn't break the data model?

**Comment by kosack** [comment]
I think my main issue here is that I want this instrument code to be used outside of ctapipe and it's I/O libraries - already I'm trying to convince the simulation configuration people to use it.  If has too many magic numbers defined in it, the data model becomes too closely tied to ctapipe or simtel.

**Comment by maxnoe** [comment]
@kosack I think you missed that the enum has string values compatible to the string values we had before, this breaks nothing, this only changes in ctapipe internally.

Observe that:
* the tests are not changed
* the data in ctapipe-extra is not changed
* Everything still works

> I also see no reason to support "circle" shapes - we always have Winston cones, so the actual area is a hexagon.

Simtel supports them and some of the cameras defined in ctapipe extra reallly have circular pixels on a hexgrid: https://www.mpi-hd.mpg.de/hfm/CT/tel3.gif

File: ctapipe/instrument/camera/geometry.py (hunk 1 above)
Comment (maxnoe) [reply ctapipe/instrument/camera/geometry.py:422]: This is an `Enum` with str values compatible to the ones we had before. Not an `IntEnum`.

**Comment by maxnoe** [comment]
This will prevent typos / enable auto completion. This is a major benefit when developing.

File: ctapipe/instrument/camera/geometry.py (hunk 1 above)
Comment (kosack) [reply ctapipe/instrument/camera/geometry.py:422]: ah, I thought they were integer values. Sorry, it does what I suggested then.

**Comment by kosack** [comment]
Ok, sorry I missed that the enum had string values. It's ok then, since it doesn't change the data model.

**Review by kosack — APPROVED** [review]

**Review by watsonjj — APPROVED** [review]

**Pull request merged**